File: pmcjs/scripts/xbase.pl

use XBase;
my $tbl = new XBase('addresses.dbf');
my $n = $tbl->last_record;
for (my $i = 0; $i <= $n; ++$i) {
  my $rec = $tbl->get_record_as_hash($i);
  print("record $i\n");
  my $k;
  foreach $k (keys %$rec) {
    my $v = $rec->{$k};
    print(" $k: $v\n");
  }
}
$tbl->close;

File: pmcjs/samples/xbase.js

use('Perl');
Perl.Use('XBase');
try {
  var tbl = new Perl('XBase', 'addresses.dbf');
} catch (e) {
  throw Perl.Methods['XBase'].errstr();
}
var n = tbl.last_record();
for (var i = 0; i <= n; ++i) {
  var rec = tbl.get_record_as_hash(i);
  print('record ' + i);
  for (var j in rec) {
    print(' ' + j + ': ' + rec[j]);
  }
}
tbl.close();

